packages/go_router/lib/src/route_data.dart

```diff
@@ -40,6 +40,10 @@ abstract class GoRouteData {
   /// [Page] implementation to be used with the results of [build].
   Page<void> buildPage(BuildContext context) => const NoOpPage();
 
+  /// A page builder for that expose the [state].
+  Page<void> buildPageWithState(BuildContext context, GoRouterState state) =>
```

Comment:
This should return buildPage by default

Reply:
Nice, I did it in https://github.com/flutter/packages/pull/2444/commits/b67e5f8fa369ad12631b072f325416a5881136bc